test/integration/testExecute.js

```diff
@@ -163,180 +165,241 @@ describe('Execute test', function ()
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+          attributeNamePrefix: TEST_ATTRIBUTE_CUSTOM_PREFIX,
+          assertionCheck: (row) => {
+            assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER], TEST_XML_VAL);
+            assert.equal(row[TEST_COL]['node'][TEST_HEADER][TEST_ATTRIBUTE_CUSTOM_PREFIX + TEST_ATTRIBUTE_NAME], undefined);
+          }
+        },
+        {
+          name: 'xml_attribute_only',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}></${TEST_HEADER}></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: TEST_ATTRIBUTE_CUSTOM_PREFIX,
+          assertionCheck: (row) => {
+            assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+            assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][TEST_ATTRIBUTE_CUSTOM_PREFIX + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+          }
+
+        },
+        {
+          name: 'xml_attribute_empty_value',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}/></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: TEST_ATTRIBUTE_CUSTOM_PREFIX,
+          assertionCheck: (row) => {
+            assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+            assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][TEST_ATTRIBUTE_CUSTOM_PREFIX + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+          }
+        }
+      ];
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      testCase.assertionCheck(row);
+    };
+
+    testCases.forEach(testCase=> it(testCase.name, createItCallback(testCase, rowAsserts)));
+  });
+
+  describe('Variant JSON', function () {
+    const TEST_JSON_VAL = '<123>';
+
+    before(async () => {
+      const createFileFormatVariant = `CREATE OR REPLACE FILE FORMAT ${TEST_VARIANT_FORMAT} TYPE = JSON`;
+      await testUtil.executeCmdAsync(connection, createFileFormatVariant);
+    });
+    //
+    beforeEach(async () => {
+      // Drop stage
+      await testUtil.executeCmdAsync(connection, createStageVariant);
+
+    });
+    afterEach(async () => {
+      // Drop stage
+      await testUtil.executeCmdAsync(connection, truncateTableVariant);
+      await testUtil.executeCmdAsync(connection, dropStageVariant);
+    });
+
+    const testCases =
+      [
+        {
+          name: 'raw_json',
+          type: 'JSON',
+          fileExtension: '.json',
+          sampleData: `{${TEST_HEADER}: \"${TEST_JSON_VAL}\"}`
+        }
+      ];
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      if (testCase.name === 'raw_json') {
```

Comment:
there is only one test case - do we need it?

Reply:
fixed